### PR TITLE
Ensure local headers are used over system headers to avoid breakage

### PR DESCRIPTION
**Pull request by chewi**
types.h was being picked up from webp rather than locally due to the `include_directories` call for GTK (and therefore its `-I` arguments) coming before the same call for the local sources. webp can be pulled in via GTK -> gdk-pixbuf -> tiff -> webp.

This can be avoided by specifying `SYSTEM` or `BEFORE` as appropriate when calling `include_directories`. I have done both for good measure.

### CODE REVIEW

File: src/CMakeLists.txt
Comment (madame-rachelle) [inline src/CMakeLists.txt:386]: We cannot use the system ASMJIT which I am assuming this implies. Our's is a modified version that is incompatible with the system libraries version.

**Review by madame-rachelle — CHANGES_REQUESTED** [review]

File: src/CMakeLists.txt
Comment (chewi) [reply src/CMakeLists.txt:386]: This doesn't change where it looks for the ASMJIT headers, it just means that your other headers will take priority in the event of a name clash, which is what you want. See `man gcc`:

```
The lookup order is as follows:

           1.  For the quote form of the include directive, the directory of the current file is searched first.

           2.  For the quote form of the include directive, the directories specified by -iquote options are searched in left-to-right order, as they appear on the command line.

           3.  Directories specified with -I options are scanned in left-to-right order.

           4.  Directories specified with -isystem options are scanned in left-to-right order.

           5.  Standard system directories are scanned.
```

Comment (madame-rachelle) [reply src/CMakeLists.txt:386]: Doesn't help that the CMAKE documentation is so vague on this. If that's the case I'll simply wait for a second opinion on this merge and dismiss the change request.

Comment (DarthGandalf) [reply src/CMakeLists.txt:386]: Cmake documentation is vague on purpose, because it targets multiple compilers :(

But for gcc the `SYSTEM` keyword means whether the directory will be passed as `-I` or `-isystem`, which then has the effect described above.

This PR doesn't reenable `find_package( asmjit )` so it won't accidentally use the system version of asmjit, especially because directories from `-isystem` are still checked before `/usr/include`

**Pull request merged**
